Add files via upload

dobrynov opened the pull request.

dobrynov committed:
Add files via upload

akuznetsov-practicum reviewed (changes requested):
Я думаю, тут можно кое-что улучшить.

У тебя ввод и вывод распределены по всему коду. Это не очень хорошо, в том плане что ты и жёстко прибиваешь в программе, что она читает только с stdin, а выводит только в stdout, и в плане что если что-то поменять, то менять нужно будет в совсем неожиданных местах.

Второе, что я думаю можно улучшить, это сделать методы ReadUsePage() и CheerUse() публичными. Так твой код можно будет переиспользовать в других приложениях, а функцию чтения данных с stdin вообще убрать за пределы класса. Так ты получишь гораздо больше гибкости в плане что кто-то может захотеть вводить данные из другого массива, например, а не stdin.

dobrynov committed:
Update booc.cpp

внес изменения, поток ввода вынес во внешний метод.
приватные методы оставил, т к подразумевается отсутствие к ним доступа вне класса

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,72 @@
+};
+
+
+int main() {
+	Book book;
+	book.Read(cin);
+}
+
+
+int main() {
+	Book book;
+	book.Read();
+}
```
Две функции `main()`? :)

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,72 @@
+		}
+	}
+private:
+	void ReadUsePage(int use_id, int page) {
```
Эту функцию лучше сделать публичной, так класс можно будет использовать в других приложениях, а не привязываться жёстко к чтению из потока ввода.

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,72 @@
+		}
+
+	}
+	void CheerUse(int use_id) {
```
Эту функцию лучше сделать публичной, так класс можно будет использовать в других приложениях, а не привязываться жёстко к чтению из потока ввода.

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,72 @@
+			cout << 1 << endl;
+		}
+		else {
+			cout << setprecision(6) << 1.0 - static_cast<double>(page_read_use_[use_page_[use_id]] - 1) / (peaple_ - 1) << endl;
```
Лучше ничего не выводить в поток из метода. Если кто-то будет использовать этот класс, они могут не ожидать, что ты будешь что-то кидать в stdout. Лучше изменить сигнатуру метода и сделать так, чтобы он возвращал `double` вместо `void`, а сам результат печатать где-нибудь за пределами класса.

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,72 @@
+public:
+	Book() : page_read_use_(1001, 0) {};
+		
+	void Read(istream& in) {
```
Этот метод лучше превратить в функцию и вынести за пределы класса. Так можно будет разделить ввод и обработку данных.

akuznetsov-practicum reviewed (changes requested):
Еще есть над чем поработать :)

dobrynov committed:
Update booc.cpp

готово!

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,68 @@
+		}
+
+	}
+	double CheerUse(int use_id) {
```
У тебя метод с возвращаемым типом, он всегда что-то должен возвращать. Некоторые компиляторы могут ругаться на то, что нет явного возврата из функции.

Можно убрать последнее условие `else` и всегда возвращать этот результат, потому что если будет какое-то другое условие, то выполнение завершиться раньше.

akuznetsov-practicum commented on boock-reading/booc.cpp:
```diff
@@ -1 +1,68 @@
+
+};
+
+void Read(istream& in, Boock boock) {
```
вот тут происходит копирование объекта `boock` , и дальше ты работаешь уже только с копией этого объекта, а должно быть использование уже имеющегося.

dobrynov committed:
Update booc.cpp

готово

akuznetsov-practicum reviewed (approved):
Теперь всё супер!
